### PR TITLE
CCD-3424 set systemTTL when TTL field is missing from case data

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [CCD-3424](https://tools.hmcts.net/jira/browse/CCD-3424) _"systemTTL field only updated if case has a TTL field"_

### Change description ###

Fix: systemTTL field only updated if case data has a TTL field.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature [hunk 1]
```diff
@@ -17,6 +17,20 @@ Feature: F-1015: Update Case - Start Case Event - Update Code for TTL
...
     And   the response [contains the TTL.SystemTTL for the case, that has been set to 20 days from today]
 
+  @S-1015.1.repeat 
+  Scenario: TTLIncrement is set to "20" for the Case Event and Start Event is invoked on v1_external#/case-details-endpoint/startEventForCaseworkerUsingGET
+    Given a user with [an active profile in CCD],
+    And   a successful call [to create a case] as in [F-1015_CreateCasePreRequisiteCaseworker_noTTL]
+    When  a request is prepared with appropriate values,
+    And   the request [contains correctly configured event details]
+    And   the request [has a TTLIncrement of 20 days configured]
+    And   the request [is configured to trigger an About To Start callback that does not change of the TTL.suspended or TTL.OverrideTTL or TTL.SystemTTL]
```
Comment (sullivangareth) [inline src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature:27]: that does not change *any*

Comment (mattnayler) [reply src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature]: Resolved in 5af480c67d8ab643d8553b812272b8f7ec1932a4

File: src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature [hunk 2]
```diff
@@ -30,6 +44,20 @@ Feature: F-1015: Update Case - Start Case Event - Update Code for TTL
...
     And   the response [contains the TTL.SystemTTL for the case, that has been set to 20 days from today]
 
+  @S-1015.2.repeat
+  Scenario: TTLIncrement is set to "20" for the Case Event and Start Event is invoked on v1_external#/case-details-endpoint/startEventForCitizenUsingGET
+    Given a user with [an active profile in CCD],
+    And   a successful call [to create a case] as in [F-1015_CreateCasePreRequisiteCitizen_noTTL]
+    When  a request is prepared with appropriate values,
+    And   the request [contains correctly configured event details]
+    And   the request [has a TTLIncrement of 20 days configured]
+    And   the request [is configured to trigger an About To Start callback that does not change of the TTL.suspended or TTL.OverrideTTL or TTL.SystemTTL]
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature]: Resolved in 5af480c67d8ab643d8553b812272b8f7ec1932a4

**Review by sullivangareth — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java [hunk 3]
```diff
@@ -27,30 +33,78 @@ public class TimeToLiveService {
...
+        }
+
+        return clonedDataClassification;
     }
 
     public Map<String, JsonNode> updateCaseDetailsWithTTL(Map<String, JsonNode> data,
-                                                          CaseEventDefinition caseEventDefinition) {
+                                                          CaseEventDefinition caseEventDefinition,
+                                                          CaseTypeDefinition caseTypeDefinition) {
         Map<String, JsonNode> clonedData = data;
```
Comment (sem-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java:87]: Same as L58 above?

**Review by sem-hmcts — COMMENTED** [review]
I've added a few comments otherwise, looks good

File: src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java [hunk 4]
```diff
@@ -27,30 +33,78 @@ public class TimeToLiveService {
...
+    public boolean isCaseTypeUsingTTL(@NonNull CaseTypeDefinition caseTypeDefinition) {
+        return Optional.ofNullable(caseTypeDefinition.getCaseFieldDefinitions()).orElse(Collections.emptyList())
+            .stream().anyMatch(caseFieldDefinition -> TTL_CASE_FIELD_ID.equals(caseFieldDefinition.getId()));
+    }
+
+    public Map<String, JsonNode> updateCaseDataClassificationWithTTL(Map<String, JsonNode> data,
+                                                                     Map<String, JsonNode> dataClassification,
+                                                                     CaseEventDefinition caseEventDefinition,
+                                                                     CaseTypeDefinition caseTypeDefinition) {
+        Map<String, JsonNode> clonedDataClassification = dataClassification;
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java]: Resolved in 51e789c546f14817e16520f57e0d91fae3596a8f

File: src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java (hunk 3 above)
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java]: Resolved in 51e789c546f14817e16520f57e0d91fae3596a8f

File: src/test/java/uk/gov/hmcts/ccd/domain/service/startevent/DefaultStartEventOperationTest.java [hunk 5]
```diff
@@ -81,9 +81,12 @@ public class DefaultStartEventOperationTest {
     private static final JsonNodeFactory JSON_NODE_FACTORY = new JsonNodeFactory(false);
     private static final int TTL_INCREMENT = 20;
 
-    private static Map<String, JsonNode> NEW_FIELDS_CLASSIFICATION =
+    private static final Map<String, JsonNode> NEW_FIELDS_CLASSIFICATION =
         ImmutableMap.of("key", JSON_NODE_FACTORY.textNode("value"));
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/startevent/DefaultStartEventOperationTest.java]: Resolved in 51e789c546f14817e16520f57e0d91fae3596a8f

**Review by sem-hmcts — APPROVED** [review]

**Pull request merged**